### ISSUE TITLE
[tone] [kagura] Sporadic random reboots

### ISSUE THREAD

**Issue by ix5** [sonyxperiadev/bug_tracker#435]
**Platform:** Tone
**Device:** Kagura
**Kernel version:** 4.9.182
**Android version:** 9.0 Pie

**Description**
Device reboots sporadically when awoken from screen off.
State: https://github.com/sonyxperiadev/kernel/pull/2016 and https://github.com/sonyxperiadev/vendor-sony-oss-fingerprint/pull/51 and https://github.com/sonyxperiadev/vendor-sony-oss-fingerprint/pull/49 merged in.

**Symptoms**
Screen stays black for ~15s
(I'm pretty sure there's a vibration)
Device shows "Your device is unlocked" screen so there must've been a crash and shutdown
Device reboots

`/sys/fs/pstore` shows no traces of any crash or kernel panic. With persistent logcat enabled, the last entries in `/data/misc/logd` are:
```
--------- switch to main
06-28 19:58:17.031836   679   836 D FPC IMP : Image capture result: 0
--------- switch to system
06-28 19:58:17.054959  1105  1130 I DisplayPowerController: Blocking screen on until initial contents have been drawn.
--------- switch to main
06-28 19:58:17.076157   679   836 D FPC IMP : Print identified as 0
06-28 19:58:17.076777   679   836 I AOSP FPC HAL (Binder): process_auth : Auth step = 0
06-28 19:58:17.077336   679   836 I AOSP FPC HAL (Binder): process_auth : Got print id : 0
--------- switch to events
06-28 19:58:17.094354  1105  1105 I screen_toggled: 1
06-28 19:58:17.094816  1105  1105 I power_screen_broadcast_send: 1
06-28 19:58:17.096255  1105  1105 I sysui_action: [252,false]
06-28 19:58:17.096603  1105  1105 I sysui_multi_action: [757,252,758,4,759,0]
--------- switch to main
06-28 19:58:17.099885  1105  1105 W VibratorService: Failed to perform effect (1)
--------- switch to events
06-28 19:58:17.104682  1105  1105 I battery_saving_stats: [0,0,1,345230,0,0,8312803,0,3]
--------- switch to main
06-28 19:58:17.108047   544   544 E WifiHAL : wifi_get_link_stats: requestResponse Error:-3
06-28 19:58:17.110821  1105  1195 E WifiVendorHal: getWifiLinkLayerStats(l.937) failed {.code = ERROR_NOT_SUPPORTED, .description = }
--------- switch to events
06-28 19:58:17.112994  1105  1105 I device_idle: [0,screen]
06-28 19:58:17.113269  1105  1105 I device_idle_light: [0,screen]
--------- switch to main
06-28 19:58:17.114506   539   539 I RQBalance-PowerHAL: Device is awake.
06-28 19:58:17.115710   539   539 I RQBalance-PowerHAL-Hints: Setting balanced mode
06-28 19:58:17.118641   539 21763 I RQBalance-PowerHAL-Hints: PowerServer is waiting for connection...
--------- switch to events
06-28 19:58:17.119313  1105  1119 I device_idle_off_start: screen
```

**How to reproduce**
On battery, no cables connected
Fingerprint enrolled and fingerprint auth enabled
GPS on
Let the device idle for a few minutes to a few hours with screen off
Put your finger on the fingerprint reader(without triggering a  power button press) to try to wake up the device

**Ideas**
Might have something to do with the TrustZone since most other components would either trigger a kernel panic that would dump logs to the persistent RAM area or would trigger a write to `/data/misc/logd`.

Possible related:
https://github.com/sonyxperiadev/bug_tracker/issues/418
https://github.com/sonyxperiadev/bug_tracker/issues/433
https://github.com/sonyxperiadev/bug_tracker/issues/421

**Comment by duncanguthrie** [comment]
For the record - I can reproduce this in one of two ways:
1. When unlocking the filesystem, just waiting is sufficient to cause it to reboot. This takes around 15-20 seconds.
2. Connecting a USB DAC and then disconnecting it also causes this behaviour to occur.
3. I also found it would occur after leaving it overnight anticipating an alarm - it would reboot and then would revert to charging from the power supply, and naturally the alarm would never ring.

**Comment by oshmoun** [comment]
loire does not crash when unlocked using the fingerprint scanner, or it didn't use to do that before. been a while since I last checked.
however, #442 looks related.

**Comment by samoylenko** [comment]
Hi All, given that Tone just got moved to the legacy section, and the plans to keep it on 4.9 binaries - just wanted to ask about this issue. For the past few months I've tried to run AOSP 9 on my Kagura, and while I had lots of help, I still gave up and rolled back to stock (android 8) because of those sporadic reboots described here. Is there any hope to get the device/platform stable on at least one of AOSP versions? I'd be happy even with AOSP 9
